### DISCUSSION TITLE
[19.10.2024] Blog 2 - Softwareanforderungsspezifikation

### DISCUSSION THREAD

**Discussion by bunki1224** [bunki1224/Softwareengineering-Projekt#19] [Show and tell]
## Einleitung

Die Reiseplaner-Webapp ist die ideale Lösung für alle, die ihre Reisen individuell und effizient gestalten möchten. Sie bietet zahlreiche Funktionen, um vollständige Reisepläne zu erstellen, die auch offline verfügbar sind. Ob Gelegenheitsreisende oder Vielreisende – mit dieser App wird die Planung deutlich erleichtert und flexibler.

### Funktionale Anforderungen

- **Reiseplanung:** Die Webapp ermöglicht es, Reisen mit mehreren Zielen und Etappen zu planen. Transportmöglichkeiten, Unterkünfte und Aktivitäten können einfach integriert werden. Alles in wenigen Klicks, was die Reiseplanung übersichtlich und stressfrei macht.
- **Offline-Zugriff:** Ein herausragendes Feature ist die Möglichkeit, den Reiseplan auch offline abzurufen und zu bearbeiten. Sobald die Internetverbindung wiederhergestellt ist, werden alle Änderungen synchronisiert. Dies ist besonders praktisch für Reisen in Regionen mit schlechter Netzabdeckung.
- **Benutzerfreundlichkeit:** Die intuitive Oberfläche sorgt dafür, dass auch unerfahrene Nutzer ihre Reisen schnell planen können. Wichtige Informationen wie Reisedaten, Unterkünfte und Sehenswürdigkeiten lassen sich unkompliziert über eine visuelle Eingabemaske hinzufügen.

### Nicht-funktionale Anforderungen

- **Sicherheit:** Alle Benutzerdaten werden sicher und verschlüsselt gespeichert. So können sich Nutzer darauf verlassen, dass ihre persönlichen Daten und Reiseinformationen gut geschützt sind.
- **Performance:** Die App liefert schnelle Ergebnisse, wobei Suchabfragen und API-Aufrufe in weniger als zwei Sekunden erfolgen. Diese Schnelligkeit sorgt für eine reibungslose Benutzererfahrung, selbst bei umfangreichen Reiseplänen.
- **Zuverlässigkeit:** Mit einer garantierten Betriebszeit von 99,9 % können sich Nutzer darauf verlassen, dass die App während der Hauptreisezeiten, insbesondere zwischen Juni und August, nahezu ohne Ausfallzeiten zur Verfügung steht.

### Dashboard
Die Mockups der Reiseplaner-Webapp zeigen eine benutzerfreundliche und übersichtliche Oberfläche, die es Nutzern leicht macht, ihre Reisen zu planen. Auf der linken Seite finden sich Suchfunktionen und Informationen zu Sehenswürdigkeiten, Restaurants und Hotels, inklusive Preisangaben und Bewertungen. Besonders praktisch ist die Integration einer Karte, die wichtige Orte und Routen visuell darstellt. Das Farbschema ist nicht final und wird in naher Zukunft geändert 
<img width="680" alt="Screenshot 2024-10-19 113310" src="https://github.com/user-attachments/assets/080f6cac-5284-4f71-9284-6a4c1eabbfee">
### Timeline
Eine Timeline ermöglicht es Nutzern, ihre Reisetage detailliert zu planen und Aktivitäten nach Datum zu ordnen. Das Ordnen geschieht über eine Drag and Drop Funktion.

<img width="680" alt="Screenshot 2024-10-19 114126" src="https://github.com/user-attachments/assets/422cb753-761c-47d0-a6a4-55697f8240e7">

> Euer TripAhead Team (JB)

**Comment by ktoffel999** [comment]
Euer Blog zeigt, dass ihr ein durchdachtes Konzept für eine Reiseplaner-Webapp entwickelt habt, das sich sowohl auf Benutzerfreundlichkeit als auch auf eine effiziente Organisation konzentriert. Die klare Strukturierung der Anforderungen und die Verwendung visueller Mockups verdeutlichen, wie die App nicht nur praktisch, sondern auch intuitiv gestaltet werden soll. Besonders beeindruckend ist, wie ihr die Integration von Echtzeitdaten und Karten visualisiert, was den Planungsprozess erheblich erleichtert. Eure Idee, Reisen individuell zu gestalten und dabei auf wichtige Tools wie eine Timeline und ein übersichtliches Dashboard zu setzen, zeigt, dass ihr die Bedürfnisse moderner Reisender gut versteht und umsetzt.

Viele Grüße 
Das Jump and Run team

**Comment by Ninetilt** [comment]
**Hey Trip Ahead Team**, wir finden euren Post echt gelungen. An euren Anforderungen kann man erkennen, dass ihr schon ein klares Bild davon hat, wie euer Reiseplaner funktionieren soll. Besonders den Offline-Zugriff halten wir für eine gute Idee, weil man auf vielen Reisen doch hin und wieder in Funklöcher fällt.

Euer UI-Mockup sieht auch super aus. Schon auf den ersten Blick kann man erkennen wofür die einzelnen Elemente stehen und man kann auch schon gut erahnen wie die App zu bedienen ist.

Uns würde noch interessieren wie die UML-Anwendungsfalldiagramme zu eurem Projekt aussehen.

Viele Grüße
Projekt TDC (JS)

**Reply by Kartoffeleintopf** [reply]
Hey Projekt TDC Team,
danke der Nachfrage!

![UML](https://github.com/user-attachments/assets/76a0b490-1683-4ce7-828a-3ada21807b63)

> Euer TripAhead Team (CN)

**Comment by pplayer1712** [comment]
Euer Konzept für die Reiseplaner-Webapp überzeugt durch eine durchdachte Kombination aus Funktionalität und Nutzerfreundlichkeit. Die Möglichkeit, Reisen mit mehreren Zielen und Etappen einfach zu planen, ist ideal für all jene, die eine flexible und individuelle Planung bevorzugen. Besonders beeindruckend ist das Offline-Feature, das den Zugriff auf Reisepläne auch ohne Internetverbindung ermöglicht und so eine hohe Flexibilität bietet.

Die visuelle Darstellung der Reisedaten, inklusive Karte und Bewertungen, macht den Planungsprozess nicht nur übersichtlicher, sondern auch informativer. Die Timeline mit Drag-and-Drop-Funktion hebt die Benutzererfahrung auf ein neues Level und zeigt, dass ihr die Wünsche der Reisenden in den Mittelpunkt stellt. Mit eurer Fokussierung auf schnelle Performance und Datensicherheit schafft ihr zudem ein hohes Maß an Zuverlässigkeit und Vertrauen.

Viele Grüße euer UMOC Team
/PL /JMS
